src/components/bc-pay-button.ts

```diff
@@ -0,0 +1,100 @@
+import {PropertyValues, html} from 'lit';
+import {customElement, property, state} from 'lit/decorators.js';
+import {BitcoinConnectElement} from './BitcoinConnectElement.js';
+import {bcIcon} from './icons/bcIcon.js';
+import {withTwind} from './twind/withTwind.js';
+import {loadingIcon} from './icons/loadingIcon.js';
+import {launchPaymentModal} from '../api.js';
+import './bc-balance.js';
+import {SendPaymentResponse} from '@webbtc/webln-types';
+import {checkIcon} from './icons/checkIcon.js';
+
+/**
+ * A button that when clicked launches a modal to pay an invoice.
+ */
+@customElement('bc-pay-button')
+export class PayButton extends withTwind()(BitcoinConnectElement) {
+  @property()
+  override title = 'Pay Now';
+
+  @property()
+  invoice?: string;
+
+  @property({})
+  preimage?: string;
+
+  @state()
+  _waitingForInvoice = false;
```

Comment:
what is waiting for invoice?

Reply:
if the button is clicked when the invoice attribute is not set, like how it works on https://bitcoin-connect.com now. I'll add a comment.